packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -176,6 +176,45 @@ abstract contract CrossDomainMessenger is
      */
     event FailedRelayedMessage(bytes32 indexed msgHash);
 
+    /**
+     * @notice Modifier for a per-message reentrancy guard.
```

Comment:
Preference to hide this functionality into a library or abstract contract that can be tested independently. Also what about memory safe solidity here?

Reply:
Much rather prefer not having a modifier at all, just put the hash into a mapping and check if the hash is in the mapping already, then clear the mapping afterwards to return the leftover gas.